src/timeMachine/components/TimeMachine.tsx

```diff
@@ -26,11 +26,15 @@ const TimeMachine: FunctionComponent = () => {
     'time-machine--split': isViewingVisOptions,
   })
 
-  let bottomContents: JSX.Element = null
-  if (activeTab === 'alerting') {
-    bottomContents = <TimeMachineAlerting />
-  } else if (activeTab === 'queries') {
-    bottomContents = <TimeMachineQueries />
+  const [bottomRef, setBottomRef] = useState(null)
+
+  const bottomContents = (): JSX.Element => {
+    if (activeTab === 'alerting') {
+      return <TimeMachineAlerting />
+    } else if (activeTab === 'queries') {
+      const bottomContentsMaxHeight = bottomRef?.getBoundingClientRect().height
```

Comment:
I think if you use `useRef`, this will change to `bottomRef.current.getBoundingClientRect().height`